### PR TITLE
[BACKLOG-35759] Fix for failing unit test. Reset metaWithFieldOptions…

### PR DESCRIPTION
**Pull request by urena-kenny**
… in TextFileOutputTest.java

### CODE REVIEW

File: engine/src/test/java/org/pentaho/di/trans/steps/textfileoutput/TextFileOutputTest.java [hunk 1]
```diff
@@ -417,6 +417,7 @@ private List<Throwable> doOutput( TextFileField[] textFileField, List<Object[]>
       .calcMetaWithFieldOptions( any( TextFileOutputData.class ) );
     when( stepMockHelper.processRowsStepMetaInterface.getMetaWithFieldOptions() ).thenCallRealMethod();
 
+    stepMockHelper.processRowsStepMetaInterface.metaWithFieldOptions = null;
```
Comment (peterrinehart) [inline engine/src/test/java/org/pentaho/di/trans/steps/textfileoutput/TextFileOutputTest.java:420]: Add a comment in here regarding why this is necessary; the step meta code is being executed multiple times without being fully re-initialized, and the step code expects this field to be null when processing the first row during execution.

**Review by peterrinehart — CHANGES_REQUESTED** [review]

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 18m 36s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 16m 33s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 19m 13s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**